src/timeMachine/actions/queryBuilderThunks.ts

```diff
@@ -321,10 +322,15 @@ export const selectTagValue = (index: number, value: string) => (
     currentTag.key === '_field'
   ) {
     newValues = [value]
+  } else if (isFlagEnabled('newQueryBuilder') && index === 0) {
+    newValues = [value]
   } else {
     newValues = [...values, value]
   }
 
+  if (newValues.length === 0 && index === 0) {
+    dispatch(removeTagSelector(index + 1))
+  }
```

Comment:
this condition removes the following column if the measurement has been selected and deselected. I wasn't sure if we should just reset the entire selection list, but settled on this since it emptying the list might lead to valid queries being invalidated, and this just cleans up a potentially empty query